fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -754,6 +755,9 @@ protected final void setupDruidDimensionsLoader(
                 throw new IllegalStateException(e);
             }
         } else if (BardFeatureFlag.DRUID_CACHE.isOn()) {
+            LOG.warn("Cache V1 feature flag is deprecated," +
+                    "use the new configuration parameter to set desired caching strategy"
+            );
```

Comment:
You should check whether or not the the underlying system property is set, not whether the feature flag is on.  And you should check both parameters.